src/ProgressMeter.jl

```diff
@@ -330,19 +359,44 @@ function updateProgress!(p::ProgressUnknown; showvalues = (), truncate_lines = f
         return
     end
 
-    if t > p.tlast+p.dt
-        dur = durationstring(t-p.tfirst)
-        msg = @sprintf "%s %d \t Time: %s" p.desc p.counter dur
-        move_cursor_up_while_clearing_lines(p.output, p.numprintedvalues)
-        printover(p.output, msg, p.color)
-        printvalues!(p, showvalues; color = valuecolor, truncate = truncate_lines)
-        flush(p.output)
-        # Compensate for any overhead of printing. This can be
-        # especially important if you're running over a slow network
-        # connection.
-        p.tlast = t + 2*(time()-t)
-        p.printed = true
-        return
+    if p.counter - p.prev_update_count >= p.check_iterations
```

Comment:
For future readability, perhaps move this check to a separate function and properly document/comment on it.